### ISSUE TITLE
[TorchInductor] _sfdp_pattern_14 misses the equivalent masked mul(scale) attention form

### ISSUE THREAD

**Issue by nikokurumi** [pytorch/pytorch#195412]
### 🐛 Describe the bug

`_sfdp_pattern_14` fuses the permuted additive-mask attention expression when the QK scores are scaled with `div(inv_scale)`, but misses the equivalent `mul(scale)` form.

## Minimal reproducer

The reproducer below is based on PyTorch's upstream `_test_sdpa_rewriter_14` test and only changes the score scaling from `div(3.0)` to `mul(1.0 / 3.0)`.

```Python
from __future__ import annotations

import torch


class Model(torch.nn.Module):
    def forward(
        self,
        query: torch.Tensor,
        key: torch.Tensor,
        value: torch.Tensor,
    ) -> torch.Tensor:
        attn_mask = torch.ones(
            query.size(1),
            key.size(1),
            dtype=torch.bool,
            device=query.device,
        ).tril(diagonal=0)

        attn_mask = attn_mask.masked_fill(
            torch.logical_not(attn_mask),
            -float("inf"),
        )

        q = query.permute(0, 2, 1, 3)
        k = key.permute(0, 2, 1, 3)
        v = value.permute(0, 2, 1, 3)

        return (
            (
                torch.matmul(q, k.transpose(-2, -1)).mul(1.0 / 3.0)
                + attn_mask
            )
            .softmax(dim=-1)
            .matmul(v)
        )


def make_case(device: torch.device):
    torch.manual_seed(1012)

    model = Model().to(device).eval()

    inputs = (
        torch.randn(
            4, 2, 16, 32,
            device=device,
            requires_grad=True,
        ),
        torch.randn(
            4, 2, 16, 32,
            device=device,
            requires_grad=True,
        ),
        torch.randn(
            4, 2, 16, 32,
            device=device,
            requires_grad=True,
        ),
    )

    return model, inputs


def main():
    device = torch.device("cuda:0")
    model, inputs = make_case(device)

    with torch.no_grad():
        eager_out = model(*inputs)

    torch._dynamo.reset()

    compiled_model = torch.compile(
        model,
        backend="inductor",
        fullgraph=True,
    )

    with torch.no_grad():
        compiled_out = compiled_model(*inputs)

    torch.testing.assert_close(compiled_out, eager_out)


if __name__ == "__main__":
    main()
```

## Observed behavior

The traces generate joint FX graphs:

```Plain
TORCH_TRACE=trace_path python test.py
tlparse trace_path -o tlparse_path
```

In my trace, the `div(3.0)` case is fused in `after_joint_graph_8.txt` into efficient SDPA:

```Plain
_scaled_dot_product_efficient_attention_default = (
    torch.ops.aten._scaled_dot_product_efficient_attention.default(
        permute_default,
        permute_default_1,
        permute_default_2,
        expand_default,
        False,
        scale=0.3333333333333333,
    )
)
getitem = _scaled_dot_product_efficient_attention_default[0]
```

In the corresponding trace for the `mul(1.0 / 3.0)` variant, the attention sequence remains unfused.

Reproduced on PyTorch main commit `e289091`.

### Versions

PyTorch: 2.15.0a0+gite289091
Python: 3.12.13
CUDA 12.9
NVIDIA RTX 4090

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by nikokurumi** [comment]
I checked the related patterns more broadly, and this appears to be a family-level coverage gap rather than something specific to _sfdp_pattern_14. The strongest related cases are _sfdp_pattern_5 and _sfdp_pattern_6. They have the same attention structure as pattern 14, but miss the corresponding mul(scale) variants. The same div-only score-scaling form also appears in other patterns: 7/8, 11/12, 15/17, and 18/19.

_sfdp_pattern_16 also looks related, since it uses the same div(inv_scale) scaling form with an additive mask and dropout. Patterns 9, 10, and 20 als have a similar div-only scaling coverage gap, though in a slightly different graph form.

This seems consistent with the basic patterns 1/2/3/4, where both scaling spellings are explicitly supported, e.g. div(inv_scale) and mul(scale). The later patterns seem to have accumulated mostly div forms without the corresponding mul forms.
